### PR TITLE
[rand gen] only request for missing shares

### PR DESCRIPTION
**Pull request by zekun000**

### CODE REVIEW

**Comment by zekun000** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/11476?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #11453** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/11453?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #11464** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/11464?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #11476** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/11476?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/aptos-labs/aptos-core/11476?utm_source=stack-comment).

File: crates/reliable-broadcast/src/lib.rs [hunk 1]
```diff
@@ -78,6 +78,18 @@ where
         <<S as BroadcastStatus<Req, Res>>::Response as TryFrom<Res>>::Error: Debug,
     {
         let receivers: Vec<_> = self.validators.clone();
+        self.multicast(message, aggregating, receivers)
+    }
+
+    pub fn multicast<S: BroadcastStatus<Req, Res> + 'static>(
```
Comment (danielxiangzl) [inline crates/reliable-broadcast/src/lib.rs:84]: nit: groupcast?

File: consensus/src/rand/rand_gen/rand_manager.rs [hunk 2]
```diff
@@ -178,28 +178,40 @@ impl<S: Share, D: AugmentedData, Storage: AugDataStorage<D>> RandManager<S, D, S
...
-            );
-            rb.broadcast(request, aggregate_state).await;
-            info!(
-                epoch = epoch,
-                round = round,
-                "[RandManager] Finish broadcasting share request",
-            );
+            tokio::time::sleep(Duration::from_millis(300)).await;
+            let maybe_existing_shares = rand_store.lock().get_all_shares_authors(&metadata);
+            if let Some(existing_shares) = maybe_existing_shares {
```
Comment (danielxiangzl) [inline consensus/src/rand/rand_gen/rand_manager.rs:194]: If existing shares are sufficient for generating decision, skip this?

**Review by danielxiangzl — APPROVED** [review]

File: consensus/src/rand/rand_gen/rand_manager.rs (hunk 2 above)
Comment (zekun000) [reply consensus/src/rand/rand_gen/rand_manager.rs:194]: if existing is enough/already decided, it returns None

**Review by zjma — APPROVED** [review]

**Pull request merged**